src/Database/Query/CacheableQueryBuilder.php

```diff
@@ -136,6 +136,11 @@ protected function runSelect()
         //and create additional identifiers
         $modelClasses = $this->getIdentifiableModelClasses($this->getIdentifiableValue());
 
+        // clear expired cached queries in tagged cache
+        if ($isTaggableStore) {
+            Cache::tags($modelClasses)->flushStale();
```

Comment:
Hey @kingyyy , thanks for the input. Interesting issue, I never heard about. Just read into one Github issue a minute ago about that. Fascinating ^^

My question would be, where this `flushStale` comes from? It is only in the `RedisTaggedStore` and non other, but `TaggableStore` is implemented also by e.g. `MemcachedStore`, which, from what I see, would just return `TaggedStore`, which does not have `flushStale`. Have you checked with other cache options?


Thanks a lot for the input! ^^

Reply:
Hi @nea ,

Thanks for catching this. You're right - I forgot to check for other cache types.

This problem occurred in one of my high-load projects using Redis, it was hitting memory limits due to accumulating stale entries.

The `flushStale` method helps mitigate this issue, but it's specific to `RedisTaggedCache`.

I'll update the PR to handle it only in `RedisTaggedCache`